worker/src/main/java/gov/cms/ab2d/worker/config/RoundRobinBlockingQueue.java

```diff
@@ -0,0 +1,357 @@
+import java.util.concurrent.locks.Condition;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.function.Predicate;
+
+@Slf4j
+public class RoundRobinBlockingQueue<E> implements BlockingQueue<E> {
+    // The list of categories
+    private List<String> categories = new ArrayList<>();
+    // The individual category queues
+    private Map<String, LinkedList<E>> categoryQueues = new HashMap<>();
+    // The current category index
+    private int currentIndex = 0;
+    // Main lock guarding all access
+    private ReentrantLock lock;
+    // Not empty condition on the lock
+    private final Condition notEmpty;
+    // Total number in the queues
+    private long count = 0;
+
+    RoundRobinBlockingQueue() {
+        log.info("Created Round Robin Blocking Queue");
+        lock = new ReentrantLock();
+        notEmpty = lock.newCondition();
+    }
+
+    // New code to purge items
+    public void removeIfTrue(Predicate<E> op) {
+        for (Map.Entry<String, LinkedList<E>> entry : categoryQueues.entrySet()) {
+            LinkedList<E> linkedList = entry.getValue();
+            if (linkedList == null || linkedList.isEmpty()) {
+                continue;
+            }
+            Iterator it = linkedList.iterator();
+            List<E> toRemove = new ArrayList<>();
+            while (it.hasNext()) {
+                E next = (E) it.next();
+                if (op.test(next)) {
+                    toRemove.add(next);
+                }
+            }
+            removeAll(toRemove);
+            if (linkedList.size() == 0) {
+                categoryQueues.remove(entry.getKey());
+                categories.remove(entry.getKey());
+            }
+        }
+    }
+
+    // New code to add items
+    public boolean add(String category, E e) {
+        log.debug("Adding {} - {}", category, e);
+        final ReentrantLock lock = this.lock;
+        lock.lock();
+        try {
+            if (!categories.contains(category)) {
+                categories.add(category);
+                notEmpty.signal();
+            }
+            Queue<E> categoryQueue = categoryQueues.computeIfAbsent(category, k -> new LinkedList<>());
+            count++;
+            return categoryQueue.add(e);
+        } finally {
+            lock.unlock();
+        }
+    }
+
+    public boolean addAll(String category, Collection<? extends E> c) {
+        final ReentrantLock lock = this.lock;
+        if (c == null || c.isEmpty()) {
+                return true;
+        }
+        lock.lock();
+        try {
+            if (!categories.contains(category)) {
+                categories.add(category);
+            }
+            Queue<E> categoryQueue = categoryQueues.computeIfAbsent(category, k -> new LinkedList<>());
+            count += c.size();
+            // c.size might be 0. Don't signal not empty until we know it's non-zero
+            if (count > 0) {
+                notEmpty.signal();
+            }
```

Comment:
So these two methods would be replaced with `addQueue(List list)` that adds another individual queue to the round-robin list.